src/templates/blog/post.js

```diff
@@ -48,17 +57,26 @@ BlogPostTemplate.propTypes = {
   date: PropTypes.instanceOf(Date).isRequired,
   html: PropTypes.string.isRequired,
   cover: PropTypes.string,
+  coverFile: PropTypes.object,
   title: PropTypes.string.isRequired,
 }
 
 export default ({ data }) => {
-  const { markdownRemark } = data
-  const { frontmatter, html } = markdownRemark
-  const { author, date, cover, title } = frontmatter
+  const { markdownRemark, coverFile } = data
+  const { fields, frontmatter, html } = markdownRemark
+  const { cover } = fields
+  const { author, date, title } = frontmatter
 
   return (
     <BlogPostTemplate
-      {...{ author, date: new Date(date), html, cover, title }}
+      {...{
```

Comment:
This style of passing props is so weird 😅 fine by me tho

Reply:
I prefer it because it is more succint and has the same readability, with no drawback, IMO ofc. The `prop={value}` notion in this case would be for the most part just repeating the same name for `prop` and `value`. Do you see any drawback in doing this?

Reply:
Nope, it's just weird to me. I honestly would buckle everything into a `blogPostProps` object and then `...blogPostProps`. It's a bit more clear. But it's up to you